ext/nio4r/bytebuffer.c

```diff
@@ -75,6 +75,7 @@ void Init_NIO_ByteBuffer()
 static VALUE NIO_ByteBuffer_allocate(VALUE klass)
 {
     struct NIO_ByteBuffer *bytebuffer = (struct NIO_ByteBuffer *)xmalloc(sizeof(struct NIO_ByteBuffer));
+    bytebuffer->buffer = NULL;
```

Comment:
And it might be good to just go ahead and zero the whole structure here, so long as you're changing stuff :wink:

Reply:
Using `realloc()` in initialize would also ensure if initialize is called repeatedly from Ruby-land no memory would be leaked.